packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -1741,7 +1741,7 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin,
   @override
   double computeMaxIntrinsicWidth(double height) {
     _layoutText();
-    return _textPainter.maxIntrinsicWidth + _caretMargin;
+    return _textPainter.maxIntrinsicWidth + math.max(0, _caretMargin);
```

Comment:
It looks like the entire `_caretMargin` getter needs to be updated instead?

Reply:
Thank you  for your review . I really thought about that initially, but my counter argument will be that no need to touch the caretMargin and allow it to be negative , this actually means a padding instead of margin . so the key point here is that negative margin should not affect the maxIntrinsicWidth means adding nothing (zero) to _textPainter.maxIntrinsicWidth  . what do you think?

Reply:
Looking at the references to `_caretMargin` in editable.dart, it's basically the space reserved for painting the caret. If we don't allow that to go negative in `computeMaxIntrinsicWidth` then we should keep that consistent everywhere else I think?

Reply:
I think you are right , I have a different  approach to abandon the idea of negative cursor width by introducing new degrees of freedom for the cursor padding and the caret margin, please review the changes